e2e_tests/odh-deployment-e2e.sh

```diff
@@ -0,0 +1,127 @@
+ sed -n "/\`\`\`/,/\`\`\`/p" |\
+  tail -n +2 | sed '$d' |\
+  sed  "s/uri:.*/uri: https:\/\/api.github.com\/repos\/$1\/odh-manifests\/tarball\/$2/" \
+  > trustyai.yaml
+
+oc apply -f trustyai.yaml
+oc apply -f trustyai-route.yaml
+oc apply -f trusty-pvc.yaml
+
+# wait for TrustyAI to spin up
+while [[ -z "$(oc get pods | grep trustyai-service | grep 1/1)" ]]
+do
+  echo "Waiting on trustyai pod to spin up"
+  sleep 5
+done
+
+
+## deploy ModelMesh ====================================================================================================
+oc new-project $MM_NAMESPACE
+oc label namespace $MM_NAMESPACE "modelmesh-enabled=true" --overwrite=true || echo "Failed to apply modelmesh-enabled label."
+oc project $MM_NAMESPACE
+
+oc apply -f secret.yaml
+oc apply -f odh-mlserver-0.x.yaml
+oc apply -f model.yaml
+
+# wait to spin up ======================================================================================================
+while [[ -z "$(oc get pods | grep modelmesh-serving | grep 5/5)" ]]
+do
+  echo "Waiting on modelserving runtime pods to spin up"
+  sleep 5
+done
+
+# get + send data to model route  ======================================================================================
+oc project $MM_NAMESPACE
+INFER_ROUTE=$(oc get route example-sklearn-isvc --template={{.spec.host}}{{.spec.path}})
+for i in {1..5};
+do
+  curl -k https://$INFER_ROUTE/infer -d @data.json
+done
+
+echo "\n Waiting for requests to appear in TrustyAI pod logs..."
+sleep 30
+
+# see if payloads are in logs  =========================================================================================
+oc project $ODH_NAMESPACE
+if [[ -z "$(oc logs $(oc get pods -o name | grep trustyai-service) | grep "Received partial input payload")" ]];
+then
+  echo "ERROR: No payload communication between ModelMesh + TrustyAI"
+  exit
+fi
+
+# grab + run data generator  ===========================================================================================
+[ -d "logger" ] && rm -Rf logger
+cp -r ../explainability-service/demo/logger logger
+TRUSTY_ROUTE=$(oc get route/trustyai-service-route --template={{.spec.host}})
+cd logger
+
+sed "s/sleep(random.randint(1, 3))/sleep(0.01)/" partial.py > partial-fast-gen.py
+SERVICE_ENDPOINT=http://$TRUSTY_ROUTE/consumer/kserve/v2 python3 partial-fast-gen.py & sleep 60 ; kill $!
+echo "\n"
+
+# setup a metric  chedule===============================================================================================
+curl --location http://$TRUSTY_ROUTE/metrics/spd/request \
+  --header 'Content-Type: application/json' \
+  --data '{
+      "modelId": "example-model-1",
+      "protectedAttribute": "input-2",
+      "favorableOutcome": {
+          "type": "DOUBLE",
+          "value": 1.0
+      },
+      "outcomeName": "output-0",
+      "privilegedAttribute": {
+          "type": "DOUBLE",
+          "value": 0.0
+      },
+      "unprivilegedAttribute": {
+          "type": "DOUBLE",
+          "value": 1.0
+      }
+  }'
+
+
```

Comment:
(Time permitting) a GET request to `http://$TRUSTY_ROUTE/metrics/spd/requests` should show that the metric request is indeed registered.